src/Admin/Ide/Ide.php

```diff
@@ -0,0 +1,581 @@
+		if ( 'graphql_page_graphql-ide' === $current_page_id ) {
+			return true;
+		}
+
+		return $is_plugin_scoped_page;
+	}
+
+	/**
+	 * Modifies the script tag for specific scripts to add the 'defer' attribute.
+	 *
+	 * This function checks if the script handle matches 'wpgraphql-ide' and, if so,
+	 * adds the 'defer' attribute to the script tag. This allows the script to be executed
+	 * after the HTML document is parsed but before the DOMContentLoaded event.
+	 *
+	 * @param string $tag    The HTML `<script>` tag of the enqueued script.
+	 * @param string $handle The script's registered handle in WordPress.
+	 *
+	 * @return string Modified script tag with 'defer' attribute included if handle matches; otherwise, unchanged.
+	 */
+	public function filter_script_loader_tag( string $tag, string $handle ): string {
+
+		if ( 'wpgraphql-ide' === $handle ) {
+			return str_replace( ' src', ' defer="defer" src', $tag );
+		}
+
+		return $tag;
+	}
+
+	/**
+	 * Update the existing GraphiQL link field configuration to say "Legacy".
+	 *
+	 * @param array<string, mixed> $field_config The field configuration array.
+	 * @param string               $field_name   The name of the field.
+	 * @param string               $section      The section the field belongs to.
+	 *
+	 * @return array<string, mixed> The modified field configuration array.
+	 */
+	public function update_graphiql_link_field_config( array $field_config, string $field_name, string $section ): array {
+		if ( 'show_graphiql_link_in_admin_bar' === $field_name && 'graphql_general_settings' === $section ) {
+			$field_config['desc'] = sprintf(
+				'%1$s<br><p class="description">%2$s</p>',
+				__( 'Show the GraphiQL IDE link in the WordPress Admin Bar.', 'wpgraphql-ide' ),
+				sprintf(
+				/* translators: %s: Strong opening tag */
+					__( '%1$sNote:%2$s This setting has been disabled by the new WPGraphQL IDE. Related settings are now available under the "IDE Settings" tab.', 'wpgraphql-ide' ),
+					'<strong>',
+					'</strong>'
+				)
+			);
+			$field_config['disabled'] = true;
+			$field_config['value']    = 'off';
+		}
+		return $field_config;
+	}
+
+	/**
+	 * Ensure the `show_graphiql_link_in_admin_bar` setting is always unchecked.
+	 *
+	 * @param mixed                $value          The value of the field.
+	 * @param mixed                $default_value  The default value if there is no value set.
+	 * @param string               $option_name    The name of the option.
+	 * @param array<string, mixed> $section_fields The setting values within the section.
+	 * @param string               $section_name   The name of the section the setting belongs to.
+	 * @return mixed The modified value of the field.
+	 */
+	public function ensure_graphiql_link_is_unchecked( $value, $default_value, string $option_name, array $section_fields, string $section_name ) {
+		if ( 'show_graphiql_link_in_admin_bar' === $option_name && 'graphql_general_settings' === $section_name ) {
```

Comment:
Not sure _why_ `$section_fields` can be a `string`, but since we dont need it, easy enough to ignore the type.
```suggestion
	public function ensure_graphiql_link_is_unchecked( $value, $default_value, string $option_name, $section_fields, string $section_name ) {
```